### PR TITLE
Improved report portal tagging

### PR DESCRIPTION
**Pull request by clacroix12**
Jira: [OCSQE-360](https://projects.engineering.redhat.com/browse/OCSQE-360)

Signed-off-by: Coady LaCroix <clacroix@redhat.com>

### CODE REVIEW

**Comment by clacroix12** [comment]
@petr-balogh I still need to add the documentation around the values we are tagging so I marked this as WIP

File: ocs_ci/framework/pytest_customization/ocscilib.py [hunk 1]
```diff
@@ -421,3 +432,58 @@ def pytest_runtest_makereport(item, call):
...
+        rp_tags.append('stage_deployment')
+    if not ocsci_config.DEPLOYMENT.get('allow_lower_instance_requirements'):
+        rp_tags.append("production")
+
+    for tag in rp_tags:
+        if tag:
+            config.addinivalue_line("rp_launch_tags", tag.lower())
+
+
+def get_ocs_csv_version():
```
Comment (vasukulkarni) [inline ocs_ci/framework/pytest_customization/ocscilib.py:469]: How about moving to utils like get_ocp_version?

Comment (clacroix12) [reply ocs_ci/framework/pytest_customization/ocscilib.py:469]: I originally had this in utils but due to some circular import issues that I didn't want to address in this PR i opted to have it here for now. I can try to move it to utils and fix the import issues but there were several that stemmed from us importing test helpers into our framework code which we shouldn't be doing.

Comment (vasukulkarni) [reply ocs_ci/framework/pytest_customization/ocscilib.py:469]: 👍  I agree on the circular import issues in utils, I am hitting that in my other PR as well

Comment (clacroix12) [reply ocs_ci/framework/pytest_customization/ocscilib.py:469]: I messed around with this and determined I can move it to `ocs_ci.ocs.resources.ocs`. I should have this PR updated shortly with this change.

**Comment by clacroix12** [comment]
@vasukulkarni I changed the function to return the `ocs_csv` instead of the version, and moved it to the `ocs` resource module. I also updated the `storage_cluster` module to utilize this function to cut down on some duplication since that is where the code originated.

I also just went ahead and optimized the imports as I had to make some additions/subtractions from them anyways.

File: ocs_ci/framework/pytest_customization/ocscilib.py [hunk 2]
```diff
@@ -421,3 +428,35 @@ def pytest_runtest_makereport(item, call):
...
+    )
+    if ocsci_config.DEPLOYMENT.get('ocs_registry_image'):
+        ocs_registry_image = ocsci_config.DEPLOYMENT.get('ocs_registry_image')
+        rp_tags.append(f"ocs_registry_image:{ocs_registry_image}")
+        rp_tags.append(f"ocs_registry_tag:{ocs_registry_image.split(':')[1]}")
+    if ocsci_config.DEPLOYMENT.get('ui_deployment'):
+        rp_tags.append('ui_deployment')
+    if ocsci_config.DEPLOYMENT.get('live_deployment'):
+        rp_tags.append('live_deployment')
+    if ocsci_config.DEPLOYMENT.get('stage'):
```
Comment (petr-balogh) [inline ocs_ci/framework/pytest_customization/ocscilib.py:455]: if you can also add one more tag for fips please will be great.  Value you can take from:  https://github.com/red-hat-storage/ocs-ci/blob/master/conf/ocsci/fips.yaml

**Comment by clacroix12** [comment]
@petr-balogh I added a tag for `fips` and rebased to resolve the merge conflict

@vasukulkarni I had to add a condition to `pytest_configure` to not execute some code that we were running during unit tests that is meant for our actual `run-ci` executions. This fixed the unit test failure I was hitting in the previous iteration. It's not the cleanest solution but it does work for now. If we want to improve how we differentiate between plain pytest executions and run-ci executions we can definitely try to do that in a separate issue. For now I'd like to get these tags into master :)

File: ocs_ci/framework/pytest_customization/ocscilib.py (hunk 2 above)
Comment (clacroix12) [reply ocs_ci/framework/pytest_customization/ocscilib.py:465]: Done!

**Comment by clacroix12** [comment]
@petr-balogh merging this now as I don't want the import changes to hit any more conflicts. I will add the doc with the tag descriptions in a follow up PR.

**Pull request merged**